### PR TITLE
WRKLDS-806: Bump kube dependencies to 1.28.2

### PR DESCRIPTION
**Pull request by ardaguclu**
This PR bumps k8s dependencies to 1.28.2 for OCP 4.15.

### CODE REVIEW

**Comment by ardaguclu** [comment]
Version failure in ipv6 is due to the origin refers to 1.27.1. 
/hold
until origin is bumped to 1.28.2

**Comment by ardaguclu** [comment]
Since the o/o bump has been merged;
/retest

**Comment by ardaguclu** [comment]
ipv6 failure is fixed by https://github.com/openshift/origin/pull/28273/

**Comment by ardaguclu** [comment]
@atiratree could you PTAL, when you have a chance?. Thanks.

File: pkg/cli/version/version.go [hunk 1]
```diff
@@ -53,18 +53,18 @@ type VersionOptions struct {
 	oClient         configv1client.ClusterVersionsGetter
```
Comment (atiratree) [inline pkg/cli/version/version.go:53]: we should update the examples as well

File: pkg/cli/cli.go [hunk 2]
```diff
@@ -143,7 +141,7 @@ func NewDefaultOcCommand(o kubecmd.KubectlOptions) *cobra.Command {
 		if kcmdutil.CmdPluginAsSubcommand.IsEnabled() {
 			// Command exists(e.g. kubectl create), but it is not certain that
 			// subcommand also exists (e.g. kubectl create networkpolicy)
-			if _, ok := allowedCmdsSubcommandPlugin[foundCmd.Name()]; ok {
+			if kubecmd.IsSubcommandPluginAllowed(foundCmd.Name()) {
```
Comment (atiratree) [inline pkg/cli/cli.go:144]: +1 for bringing this in. 

Btw, would you mind looking at the plugin completion?

https://github.com/kubernetes/kubernetes/blob/1020678366f741834f06b9f0087e57e48f93ffba/staging/src/k8s.io/kubectl/pkg/cmd/cmd.go#L329

https://github.com/kubernetes/kubernetes/blob/1020678366f741834f06b9f0087e57e48f93ffba/staging/src/k8s.io/kubectl/pkg/cmd/cmd.go#L472

It seems we are missing it.

File: pkg/cli/version/version.go (hunk 1 above)
Comment (ardaguclu) [reply pkg/cli/version/version.go:53]: Done

File: pkg/cli/cli.go (hunk 2 above)
Comment (ardaguclu) [reply pkg/cli/cli.go:144]: Updated code with the shell completion for plugins. But this https://github.com/kubernetes/kubernetes/blob/1020678366f741834f06b9f0087e57e48f93ffba/staging/src/k8s.io/kubectl/pkg/cmd/cmd.go#L472 seems to be added in 1.29 and we can get this in in next bump

Comment (atiratree) [reply pkg/cli/cli.go:144]: Ah, I see. Thanks for adding the first part.

**Pull request merged**